samcli/local/apigw/local_apigw_service.py

```diff
@@ -77,16 +127,15 @@ def create(self):
                           static_folder=self.static_dir  # Serve static files from this directory
                           )
 
-        for api_gateway_route in self.routing_list:
+        for api_gateway_route in self.api.routes:
             path = PathConverter.convert_path_to_flask(api_gateway_route.path)
-            for route_key in self._generate_route_keys(api_gateway_route.methods,
-                                                       path):
-                self._dict_of_routes[route_key] = api_gateway_route
+            route_key = self._route_key(api_gateway_route.method, path)
+            self._dict_of_routes[route_key] = api_gateway_route
 
             self._app.add_url_rule(path,
                                    endpoint=path,
                                    view_func=self._request_handler,
-                                   methods=api_gateway_route.methods,
+                                   methods=[api_gateway_route.method],
```

Comment:
Why change this from a list? I think the Routes should be modeled with a list of methods to a given function. This will help us avoid some of how flask resolves routes: https://github.com/awslabs/aws-sam-cli/issues/400#issuecomment-389397491